middleware/test/responsewriter.go

```diff
@@ -7,9 +7,11 @@ import (
 )
 
 // ResponseWriter is useful for writing tests. It uses some fixed values for the client. The
-// remote will always be 10.240.0.1 and port 40212. The local address is always 127.0.0.1 and
-// port 53.
-type ResponseWriter struct{}
+// remote will always be 10.240.0.1 and port 40212 if the RemoteAddress is not specified.
```

Comment:
A lot of other testing code is hardcoding this address because is a default here.. This change is mostly to accommodate for v6, no? I wonder if a `ResponseWriter6` with a hardcoded v6 address is a better idea here?

Reply:
Yes I thought of adding ResponseWriter6 as well.  However,  it would be a duplicate code to the ResponseWriter except for the default address.   So I took the parameterized path.   I can change to add ResponseWrite6 if this is preferable.  Please let me know.

Reply:
Yes, lets add that. You can just
~~~
type ResponseWriter6 struct {
    ResponseWriter
}
~~~
And then overwrite `LocalAddr()` and `RemoteAddr()`. Looking at the code the net.ParseIP part could also be lifted out, making those methods a 1-liner.